components/script/dom/window.rs

```diff
@@ -2141,16 +2141,35 @@ impl Window {
         y: f32,
         scroll_id: ExternalScrollId,
         _behavior: ScrollBehavior,
-        _element: Option<&Element>,
+        element: Option<&Element>,
         can_gc: CanGc,
     ) {
+        // We are recording the scroll offset here, to know whether the element is scrolled and
+        // we need to fire events related to it.
+        let initial_scroll_offset =
+            self.scroll_offset_query_with_external_scroll_id_no_reflow(scroll_id);
```

Comment:
It looks wrong to call a `no_reflow` function (which requires clean layout) before a `reflow` call

Reply:
Discussed offline. The correct things to do here are:
1. A reflow that doesn't update any scroll offsets, just to ensure clean layout
2. Calculate initial scroll offset
3. Another reflow just to update the scroll offsets. The internal implementation of `reflow` makes sure this is doing as little work as possible
4. Calculate the updated scroll offset
5. Fire scroll events if needed